### ISSUE TITLE
关于复现 ALBEF + Flickr30K adv=1 攻击流程的问题

### ISSUE THREAD

**Issue by jacknight1** [whdii/TMM#13]
您好，感谢作者开源 **Transferable Multimodal Attack on Vision-Language Pre-training Models** 的代码。

我正在尝试使用本仓库复现 **ALBEF + Flickr30K 的 adv=1 攻击流程**，但在复现过程中遇到了一些工程接口和辅助函数缺失的问题。

我的运行命令大致如下：

```bash
python EvalTransferAttack.py \
  --adv 1 \
  --gpu 0 \
  --config ./configs/Retrieval_flickr.yaml \
  --output_dir ./output/Retrieval_flickr_adv \
  --checkpoint ./checkpoints/albef_flickr.pth \
  --text_encoder ./pretrained/bert-base-uncased \
  --log_name adv_test \
  --save_json_name adv_test.json \
  --config_name Retrieval_flickr \
  --save_dir ./adv_output
```

在复现过程中，我发现以下问题：

1. `TextAttacker` 似乎需要传入 `device` 参数，但 `EvalTransferAttack.py` 中调用时没有传入；
2. `MultiModalAttacker` 中似乎没有初始化 `image_attacker`、`image_normalize`、`ssim`，并且缺少 `getGradCam`；
3. `ImageAttacker` 中调用了 `input_diversity`、`preprocess` 和 `get_kernel`，但我在当前仓库中没有找到对应实现；
4. `model_retrieval.py` 中的 `inference()` 似乎没有返回 `TextAttacker` 所需的 `weights`；
5. `getAtt()` 中使用的 attention map / attention gradients 似乎没有在模型中正确开启；
6. 当 `batch_size_test=1` 时，图像索引可能为空，导致后续攻击流程报错。

请问当前 GitHub 仓库是否是论文实验中使用的完整代码版本？
如果方便的话，能否提供复现 **ALBEF + Flickr30K adv=1** 的推荐运行命令、配置文件，或者补充相关 helper functions？

我主要想确认以下几个部分的原始实现：

* `TextAttacker` 中使用的 token importance `weights`；
* `input_diversity`；
* `getGradCam`；
* ALBEF + Flickr30K 下 `adv=1` 的推荐配置。

非常感谢！
